dockerfiles/Dockerfile.osm-healthcheck

```diff
@@ -0,0 +1,14 @@
+ARG GO_VERSION
```

Comment:
I think we should discuss the option of including the healthcheck within the sidecar container. This lets this not require its own container resources. This could be expensive for a large cluster.

* A small bonus is that users don't have to see yet another container
* Having a single sidecar for intercepting traffic makes logical sense. It's what I would expect as a user.
* It provides a platform for us to package more things in the sidecar (static configs, wasm filters, startup scripts, etc). Although I don't think this particular point should be a used in a debate on whether or not to use a single container.
* I see no concern of performance impact here. The code is tiny, will have a tiny footprint, and any erratic probe that hits this too hard would have had a similar performance impact to envoy itself.
* Other sidecars take this approach

Reply:
FWIW at injection time we can supply an arg to not run this. I'm imagining the new entrypoint is a small bash script that starts this up (conditionally) along with envoy and any provided args.

With appropriate testing I see the risk as relatively minimal. I'd be in favor of trying to determine what a desired end goal is, and once we have that decision, then circle back to risk/effort to get there. There may be larger caveats in terms of upgrading sidecars (currently users can override their sidecar version in the helm chart). And then we can weight up which path to take to move this forward. Or at least create an issue to circle back to this. Thoughts?

cc @jaellio, @shashankram, @snehachhabria, @trstringer

Reply:
I am in favor of keeping the healthcheck service a separate container. It keeps things simple, modular, and allows setting individual container resource limits for the data plane (envoy) and control plane (health check service). Data plane vs control plane separation is important. This approach has worked well so far so I suggest not tweaking this unless there is a strong motivation to do so, which I do not see with this change.

Reply:
FWIW this is moving functionality currently in Envoy (forwarding traffic -- even healthchecks) to a separate container, so I would call this part of the data plane as well. On the resource consumption, many managed K8s services have min resources. If I have say a large zookeeper cluster this could become expensive

Reply:
We can disagree on the terminology of "control plane traffic", but to me, data traffic refers to application traffic while control plane traffic refers to anything required to ensure the healthy functioning of the application itself (e.g. probes). I also disagree that this is moving functionality in Envoy to a separate container because this fix is specifically targeting a scenario Envoy cannot address for TCP ports. Given this service is lightweight, I see no issues keeping this as is. If resource consumption is a concern, add resource limits to the health check container. 

@jaellio Feel free to merge this fix after addressing the pending comments. Revamping the existing architecture to augment this with the Envoy container is a much broader discussion that we should not be blocked on.

Reply:
Agreed on not blocking this issue, but I think this requires further discussion. Kicked off #4563. I mentioned in that bug but I think it's worth pointing out again that some platforms have minimum resource limits that are non-trivial, so the cost here can still be high.